tests/unit/providers/rhel/test_alma_inheritance_cases.py

```diff
@@ -0,0 +1,511 @@
+3. AlmaLinux has corresponding advisory with specific package version (use Alma version)
+4. AlmaLinux has no corresponding advisory (version = "None", NoAdvisory = True)
+"""
+
+import copy
+import os
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from vunnel.providers.rhel import Config, Provider
+from vunnel.workspace import Workspace
+
+
+class TestAlmaInheritanceCases:
+    @pytest.fixture
+    def rhel_config_with_alma(self):
+        return Config(include_alma_fixes=True)
+
+    @pytest.fixture
+    def mock_alma_workspace_with_scenarios(self):
+        """Create mock AlmaLinux errata data covering all 4 test scenarios"""
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # Create the directory structure for HTTP-based errata files
+            rhel_input_dir = os.path.join(tmpdir, "rhel", "input")
+            alma_dir = os.path.join(rhel_input_dir, "alma-errata-data")
+            os.makedirs(alma_dir, exist_ok=True)
+
+            import orjson
+
+            # Create mock errata-8.json file with all test scenarios
+            mock_errata_8 = [
+                # Scenario 3: AlmaLinux has corresponding advisory with specific package version
+                {
+                    "updateinfo_id": "ALSA-2022:6158",
+                    "type": "security",
+                    "severity": "Moderate",
+                    "title": "Moderate: php:7.4 security update",
+                    "pkglist": {"packages": [{"name": "php", "epoch": "0", "version": "7.4.19", "release": "4.module_el8.6.0+3238+624bf8b8"}]},
+                    "references": [{"type": "rhsa", "id": "RHSA-2022:6158"}],
+                },
+                # Scenario 2: AlmaLinux has corresponding advisory but no package entry for our test package
+                {
+                    "updateinfo_id": "ALSA-2021:1809",
+                    "type": "security",
+                    "severity": "Important",
+                    "title": "Important: httpd security update",
+                    "pkglist": {
+                        "packages": [
+                            {
+                                "name": "other-package",  # Different package, not httpd
+                                "epoch": "0",
+                                "version": "1.0.0",
+                                "release": "1.el8",
+                            }
+                        ]
+                    },
+                    "references": [{"type": "rhsa", "id": "RHSA-2021:1809"}],
+                },
+                # Scenario 1: AlmaLinux-specific advisory with A prefix
+                {
+                    "updateinfo_id": "ALSA-2025:A004",
+                    "type": "security",
+                    "severity": "Critical",
+                    "title": "Critical: vulnerable-pkg security update",
+                    "pkglist": {"packages": [{"name": "vulnerable-pkg", "epoch": "0", "version": "2.1.0", "release": "1.el8.alma"}]},
```

Comment:
The fact that `.alma` means it's an alma-specific version should be commented somewhere (though ideally not here).